services/hugo_service.py

```
@@ -65,8 +69,15 @@ def start(self, debug=False):
             ]
             if debug:
                 cmd.append("-D")
-            # Docker 部署时禁用了 Hugo Modules，需要显式指定主题
+            # Docker 部署时禁用了 Hugo Modules，需要显式指定主题。
+            # 环境变量优先级高于持久化设置，避免破坏现有部署。
             theme = os.environ.get("HUGO_THEME", "")
+            if not theme and self.settings_service is not None:
+                try:
+                    settings = self.settings_service.get_settings()
+                    theme = settings.get("theme", {}).get("name", "")
+                except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)